packages/unmock-node/src/backend/index.ts

```diff
@@ -29,168 +32,159 @@ const mHttp = (
     ignore,
     whitelist,
   }: IUnmockInternalOptions,
-  cb: {
-    // tslint:disable-next-line:max-line-length
-    (
-      options: string | http.RequestOptions | URL,
-      callback?: ((res: http.IncomingMessage) => void) | undefined,
-    ): http.ClientRequest;
-    // tslint:disable-next-line:max-line-length
-    (
-      url: string | URL,
-      options: http.RequestOptions,
-      callback?: ((res: http.IncomingMessage) => void) | undefined,
-    ): http.ClientRequest;
-  },
+  req: IncomingMessage,
+  res: ServerResponse,
 ) => {
-  // Return a function that will work for the overloaded methods
-  // parameters will correspond to either of the following signatures:
-  // request(options: RequestOptions | string | URL, callback?: (res: IncomingMessage) => void): ClientRequest;
-  // request(url: string | URL, options: RequestOptions, callback?: (res: IncomingMessage) => void): ClientRequest;
-  return (
-    first: RequestOptions | string | URL,
-    second: RequestOptions | ((res: IncomingMessage) => void) | undefined,
-    third?: (res: IncomingMessage) => void,
-  ): ClientRequest => {
-    let data: {} | null = null;
-    let selfcall = false;
-    const responseData: Buffer[] = [];
-    const ro = (first instanceof URL || typeof first === "string"
-      ? second
-      : first) as RequestOptions;
-    if (hostIsWhitelisted(whitelist, ro.host, ro.hostname)) {
-      // TODO
-      // we know this will work because this is the original signature, but is there
-      // a better way to make this typesafe?
-      return cb(first as any, second as any, third as any);
-    }
-    // tslint:disable-next-line:max-line-length
-    const pathForFake = buildPath(
-      ro.headers,
-      ro.host,
-      ro.hostname,
-      ignore,
-      ro.method,
-      ro.path,
-      signature,
-      story.story,
-      unmockHost,
-      token !== undefined,
+  const { Host, ...rawHeaders } = rawHeadersToHeaders(req.rawHeaders);
+  const [h, p] = Host.split(":");
```

Comment:
Could we do `[host, port]`? It's harder to keep track of single-letter variables later on.

Reply:
The problem is that there are other variables named host and port later on. Agreed that this sucks, though...